problems/[230110] 54. Spiral Matrix.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @param {number[][]} matrix
+ * @return {number[]}
+ */
+const spiralOrder = function(matrix) {
+  if (matrix[0].length === 1 || matrix.length === 1) return matrix.flat();
+
+  const result = [];
+  const totalCount = matrix.flat().length;
+  let count = 0;
+
+  let firstIndex = 0;
+  let secondIndex = -1;
+
+  let rightDirection = true;
+  let bottomDirection = false;
+  let leftDirection = false;
+  let topDirection = false;
+
+  let rightLimitIndex = matrix[0].length - 1;
+  let bottomLimitIndex = matrix.length - 1;
+  let leftLimitIndex = 0;
+  let topLimitIndex = 1;
+
+  while (true) {
```

Comment:
일종의 switch문 재귀를 구현하신거군요..! 그러면 각 direction마다 flag를 설정해주는 변수를 만드는 것보다 하나의 변수로 작성하게 만들 수도 있지 않을까요..?

```suggestion
  function record() {
     switch (direction % 4) {
       case 0: //rightDirection
       case 1: //bottomDirection
       case 2: //leftDirection
       case 3: //topDirection {
         // ...코드 실행
         direction += 1;
       }
     }
     
     record(); // 조건이 맞을 때까지 재실행
  }
```

Reply:
우와.... 그런 방법이!!!!!!!!!
정말 생각도 못했어요. 너무너무 좋은 생각이네요. 조언 감사합니다🙇‍♀️